internal/runtimehandlerhooks/high_performance_hooks.go

```diff
@@ -204,34 +204,24 @@ func annotationValueDeprecationWarning(annotation string) string {
 // Since CRI-O is the owner of the container cgroup, it must set this value for
 // the container. Some other entity (kubelet, external service) must ensure this is the case for all
 // other cgroups that intersect (at minimum: all parent cgroups of this cgroup).
-func disableCPULoadBalancing(c *oci.Container) error {
-	lspec := c.Spec().Linux
-	if lspec == nil ||
-		lspec.Resources == nil ||
-		lspec.Resources.CPU == nil ||
-		lspec.Resources.CPU.Cpus == "" {
-		return fmt.Errorf("find container %s CPUs", c.ID())
-	}
-
+func disableCPULoadBalancing(parentDir string, c *oci.Container) error {
 	if node.CgroupIsV2() {
 		return fmt.Errorf("disabling CPU load balancing on cgroupv2 not yet supported")
 	}
 
-	pid, err := c.Pid()
+	_, containerManagers, err := libctrManagersForPodAndContainerCgroup(c, parentDir)
```

Comment:
Instantiating two cgroup managers for the sake of getting the path to cpuset cgroup seems like an overkill.

I would do it differently -- get the path to cgroup* , and in case its last component is "container", write to the parent cgroup as well.

* as for getting the path to cgroup, you can do it in the same way like it's done before this patch, OR maybe you can actually cache the cgroup path in Container struct somehow (similar to how you do it for Sandbox and CgroupParent).

Reply:
I agree the managers are heavy and complicate things. However hardcoding a name based detection is ugly and not future proof.

Reply:
> you can do it in the same way like it's done before this patch

this won't work for the sched_load_balance case because we need to disable after the container has stopped too (in case it has load balancing while  it was running but we need to reallocate the cpus it used to use later), which means the proc entry is gone